### PR TITLE
Finnish Translation

### PR DESCRIPTION
**Pull request by Hikutaatti**
WIP

### CODE REVIEW

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 1]
```diff
@@ -223,6 +239,15 @@ public enum InternalMessage {
                     "including but not limited to private messages or staff chat messages without /commands\n" +
                     "\n" +
                     "\n");
+            put(Language.FI,
+                    "\n" +
+                    "\n" +
+                    "RespectChatPlugins on asetettu pois käytöstä. Tämä asetus on VAIN TESTAUSTA VARTEN,\n" +
+                    "ja sitä ei tule KOSKAAN käyttää lopullisella palvelimella.\n" +
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:246]:
```suggestion
                    "ja sitä ei tule KOSKAAN käyttää tuotanto palvelimella.\n" +
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 2]
```diff
@@ -223,6 +239,15 @@ public enum InternalMessage {
...
+                    "Asettamalla asetuksen pois käytöstä, aiheutat peruutettujen viestien lähettämisen Discordiin,\n" +
+                    "jotka voivat sisältää yksitysviestejä tai henkilökunnan viestejä, joissa ei ole komentoja.\n" +
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:248]:
```suggestion
                    "jotka voivat sisältää yksitysviestejä tai henkilökunnan viestejä, jotka ei käytä komentoja.\n" +
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 3]
```diff
@@ -351,6 +379,7 @@ public enum InternalMessage {
             put(Language.PL, "Znaleziono serwer");
         }}), NOT_FORWARDING_CONSOLE_OUTPUT(new HashMap<Language, String>() {{
             put(Language.EN, "Console channel ID was invalid, not forwarding console output");
+			put(Language.FI, "Konsoli kanavan ID on väärin. Konsolia ei jaeta eteenpäin.")
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:382]:
```suggestion
            put(Language.FI, "Konsoli kanavan ID on väärin. Konsolia ei välitetä eteenpäin.")
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 4]
```diff
@@ -423,6 +457,7 @@ public enum InternalMessage {
             put(Language.PL, "Nie udało się pobrać adresu URL");
         }}), HTTP_FAILED_TO_DOWNLOAD_URL(new HashMap<Language, String>() {{
             put(Language.EN, "Failed to download URL");
+			åut(Language.FI, "URL osoittetta ei voitu ladata");
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:460]:
```suggestion
            put(Language.FI, "URL osoittetta ei voitu ladata");
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 5]
```diff
@@ -531,6 +574,7 @@ public enum InternalMessage {
             put(Language.PL, "Gracz {player} dołączył z uprawnieniem do cichego dołączania, bez wysyłania wiadomości o dołączeniu");
         }}), SILENT_QUIT(new HashMap<Language, String>() {{
             put(Language.EN, "Player {player} quit with silent quitting permission, not sending a quit message");
+			put(Language.FI, "Pelaaja {player} lähti käyttäen hiljaisen liittymisen oikeuksia. Ei lähetetä poistumis viestiä");
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:577]:
```suggestion
            put(Language.FI, "Pelaaja {player} lähti käyttäen hiljaisen poistumisen oikeuksia. Ei lähetetä poistumis viestiä");
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 6]
```diff
@@ -651,6 +704,7 @@ public enum InternalMessage {
             put(Language.PL, "Podczas przetwarzania polecenia wystąpił błąd wewnętrzny.");
         }}), RESYNC_WHEN_GROUP_SYNC_DISABLED(new HashMap<Language, String>() {{
             put(Language.EN, "Group synchonization requires valid GroupRoleSynchronizationGroupsAndRolesToSync entries in synchronization.yml");
+			put(Language.FI, "Ryhmien synkronisointi vaatii oikeanlaiset GroupRoleSynchronizationGroupsAndRolesToSync merkinnät synchronization.yml tiedostossa");
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:707]: "synkronisointi" -> "synkronointi"?

**Review by Vankka — COMMENTED** [review]
Tab is being used for indentation, those will need to be converted to 4 spaces (DiscordSRV uses spaces for indentation)

Added additional in-line suggestions/comments below.

**Comment by Vankka** [comment]
"Discord" & "DiscordSRV" capitalization is wrong in some places (mainly the first letter being lowercase when it should be uppercase)

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java (hunk 6 above)
Comment (Hikutaatti) [reply src/main/java/github/scarsz/discordsrv/util/LangUtil.java:707]: 👍 That is what it should be anyways.

File: src/main/resources/messages/fi.yml [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+# Used when an error occurs with permissions for the player to run the command, not an error running the command itself
+# This is sent as a PM to the user
+#
+# Available placeholders:
+# %user%:  the name of the user that tried running the command
+#           example: Notch
+# %error%: the reason for the error
+#           example: no permission
+#
+DiscordChatChannelConsoleCommandNotifyErrorsFormat: "**%user%**, yritit suorittaa komentoa. Harmillisesti törmäsimme virheeseen: %error%"
```
Comment (Vankka) [inline src/main/resources/messages/fi.yml:152]:
```suggestion
DiscordChatChannelConsoleCommandNotifyErrorsFormat: "**%user%**, yritit suorittaa komentoa. Valitettvasti tapahtui virhe: %error%"
```

File: src/main/resources/messages/fi.yml [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+# %username%:      the username of the user that is being replied to on Discord
+#                   example: Notch
+# %message%:       the content of the message that is being replied to
+#
+# NOTE: The %reply% placeholder needs to be present in the format if you want the DiscordToMinecraftMessageReplyFormat to display in your message.
+#
+DiscordToMinecraftChatMessageFormat: "[<aqua>Discord</aqua> | %toprolecolor%%toprolealias%<reset>] %name%%reply% » %message%"
+DiscordToMinecraftChatMessageFormatNoRole: "[<aqua>Discord</aqua>] %name%%reply% » %message%"
+DiscordToMinecraftAllRolesSeparator: " | "
+DiscordToMinecraftMessageReplyFormat: " (vastaa pelaajalle %name%)"
```
Comment (Vankka) [inline src/main/resources/messages/fi.yml:51]:
```suggestion
DiscordToMinecraftMessageReplyFormat: " (vastaa käyttäjälle %name%)"
```

File: src/main/resources/linking/fi.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  #
+  # This option's value is superseded when you have subscriber roles enforced below.
+  Must be in Discord server: true
+
+  # Optionally require people to not only be linked but also to have a one of or all specified roles like a Twitch sub role
+  Subscriber role:
+    Require subscriber role to join: false
+    Subscriber roles: ["00000000000000000", "00000000000000000", "00000000000000000"]
+    Require all of the listed roles: false # when false, only one of the above roles is required
+    Kick message: "&cYou must be subscribed on Twitch to be able to play."
```
Comment (Vankka) [inline src/main/resources/linking/fi.yml:50]:
```suggestion
    Kick message: "&cSinun täytyy tilata Twitch-issä jotta voit pelata."
```

File: src/main/resources/linking/fi.yml [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+  Messages:
+    DiscordSRV still starting: "&cLinkkittämis tilaa ei voida toistaiseksi tarkistaa, koska palvelin yhdistää vielä Discordiin \n\ Kokeile myöhemmin uudelleen!
+    Not in server: "&cEt ole vielä Discord palvelimellamme! \n\nLiity nyt: {INVITE}!"
+    Failed to find subscriber role: "&cTilaaja roolia ei löydetty Discordista.\n\nOta yhteys ylläpitoon!"
+    Failed for unknown reason: "&cKäyttäjääsi ei voitu vahvistaa virheen takia.\n\nOta yhteys ylläpitoon!"
+    Kicked for unlinking: "&cSinut on potkittu palvelimelta linkittämisen poiston myötä!\n\nLinkitä käyttäjäsi uudelleen päästäksesi takaisin."
```
Comment (Vankka) [inline src/main/resources/linking/fi.yml:57]:
```suggestion
    DiscordSRV still starting: "&cLinkkittämis tilaa ei voida toistaiseksi tarkistaa, koska palvelin yhdistää vielä Discordiin.\n\nKokeile myöhemmin uudelleen!"
    Not in server: "&cEt ole vielä Discord palvelimellamme.\n\nLiity nyt: {INVITE}!"
    Failed to find subscriber role: "&cTilaaja roolia ei löydetty Discordista.\n\nOta yhteys ylläpitoon."
    Failed for unknown reason: "&cKäyttäjääsi ei voitu vahvistaa virheen takia.\n\nOta yhteys ylläpitoon."
    Kicked for unlinking: "&cSinut on potkittu palvelimelta linkittämisen poiston myötä.\n\nLinkitä käyttäjäsi uudelleen päästäksesi takaisin."
```

File: src/main/resources/config/fi.yml [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+#     %totalplayers%:   total number of players to ever join the main world
+#     %timestamp%:      current unix timestamp
+#   UpdateInterval: Time in minutes to wait between updating the channel's name (minimum is 10 due to rate limits)
+ChannelUpdater:
+  - ChannelId: "0000000000000000"
+    Format: "%playercount% pelaajaa paikalla"
+    ShutdownFormat: "Palvelin on pois päältä"
+    UpdateInterval: 10
+  - ChannelId: "0000000000000000"
+    Format: "Current TPS: %tps%"
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:298]:
```suggestion
    Format: "Nykyinen TPS: %tps%"
```

File: src/main/resources/config/fi.yml [hunk 12]
```diff
@@ -0,0 +1,350 @@
...
+# Chat channel player list command
+# All the config stuff for the player list command
+#
+# DiscordChatChannelListCommandEnabled: whether the command is enabled
+# DiscordChatChannelListCommandMessage: the command people can type to get the player list
+# DiscordChatChannelListCommandExpiration: time in seconds until a sent player list message is automatically removed by the bot. set to 0 to disable expiration.
+# DiscordChatChannelListCommandExpirationDeleteRequest: whether to delete the message of the person that originally requested for the player list
+#
+DiscordChatChannelListCommandEnabled: true
+DiscordChatChannelListCommandMessage: "lista"
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:243]:
```suggestion
DiscordChatChannelListCommandMessage: "playerlist"
```

File: src/main/resources/synchronization/fi.yml [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+# Ban synchronization
+# When a player gets banned on the server when they have a linked Discord account you can optionally ban them on the Discord server and vice versa
+#
+# BanSynchronizationDiscordToMinecraft: whether to ban people on the Minecraft server if they get banned from the Discord server
+# BanSynchronizationDiscordToMinecraftReason: the message to be used as the ban reason for banning players from the Minecraft server
+# BanSynchronizationMinecraftToDiscord: whether to ban people on the Discord server if they get banned from the Minecraft server
+#
+BanSynchronizationDiscordToMinecraft: false
+BanSynchronizationDiscordToMinecraftReason: "&cSinut on toistaiseksi bännätty palvelimelta, koska olet myös bännätty discord palvelimellamme. Ota yhteys ylläpitoon jos luulet tämän olevan virhe."
```
Comment (Vankka) [inline src/main/resources/synchronization/fi.yml:53]:
```suggestion
BanSynchronizationDiscordToMinecraftReason: "&cSinut on toistaiseksi bännätty palvelimelta, koska olet myös bännätty Discord palvelimellamme. Ota yhteys ylläpitoon jos luulet tämän olevan virhe."
```

File: src/main/resources/messages/fi.yml [hunk 14]
```diff
@@ -0,0 +1,444 @@
...
+DiscordLinkedAccountCheckFailed: "Tilisi linkitys tilaa ei voitu tarkistaa. Yritä myöhemmin uudelleen."
+# Minecraft
+ClickToCopyCode: "Klikkaa kopioidaksesi"
+MinecraftAccountLinked: "&bSinun UUID:si on linkitetty Discord käyttäjään: %username% (%id%)"
+MinecraftAccountAlreadyLinked: "&bSinun Minecraft käyttäjäsi on jo liitetty Discord tiliin. Jos sinulla on tähän oikeus, voit poistaa linkityksen komennolla /discord unlink."
+LinkedCommandSuccess: "&bMinecraft käyttäjäsi on liitetty tiliin: %name%."
+UnlinkCommandSuccess: "&bMinecraft käyttäjäsi yhdistäminen on poistettu tililtä: %name%."
+MinecraftNoLinkedAccount: "&cMinecraft käyttäjäsi ei ole yhdistetty Discord tiliin."
+LinkingError: "&cTilejä ei voitu linkittää sisäisen ongelman myötä. Ota yhteyttä ylläpitoon!"
+MinecraftNobodyFound: "&cKohdetta vastaavaa Discord ID:tä/Discord nimeä/Minecraft nimeä/Minecraft UUID:tä  \"%target%\" ei löytynyt."
```
Comment (Vankka) [inline src/main/resources/messages/fi.yml:444]:
```suggestion
MinecraftNobodyFound: "&cKohdetta vastaavaa Discord ID:tä/Discord nimeä/Minecraft nimeä/Minecraft UUID:tä \"%target%\" ei löytynyt."
```

File: src/main/resources/messages/fi.yml [hunk 15]
```diff
@@ -0,0 +1,444 @@
...
+# UnlinkCommandSuccess:         %name%:         the discord username of the Discord user that the user's Minecraft account was linked to
+# MinecraftNobodyFound:         %target%:       the input that led to no results being found
+#
+# Discord
+CodeGenerated: "Linkitys koodisi on: %code%. Lähetä botille yksityisviesti (%botname%), joka sisältää vain tämän nelinumeroisen koodin."
+UnknownCode: "En tunnista antamaasi koodia. Kokeile uudestaan."
+InvalidCode: "Oletko varma että tuo on koodisi? Linkitys koodit ovat aina vain neljä numeroa."
+DiscordAccountLinked: "Discord tilisi on linkitetty seuraavaan käyttäjään: %name% (%uuid%)"
+DiscordAccountAlreadyLinked: "Sinut on jo linkitetty käyttäjälle: %username% (%uuid%)"
+DiscordLinkedAccountRequired: "Yritit lähettää viestin pelin chat kanavalle, mutta tämä palvelin vaatii, että sinun käyttäjäsi on linkitetty Discord-tiliisi.  Linkitä se pelissä kirjoittamalla `/discord link `. \n```%message%```"
```
Comment (Vankka) [inline src/main/resources/messages/fi.yml:434]:
```suggestion
DiscordLinkedAccountRequired: "Yritit lähettää viestin pelin chat kanavalle, mutta tämä palvelin vaatii, että sinun käyttäjäsi on linkitetty Discord-tiliisi. Linkitä se pelissä kirjoittamalla `/discord link `. \n```%message%```"
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 16]
```diff
@@ -399,6 +431,7 @@ public enum InternalMessage {
             put(Language.PL, "Odbiornik API {listenername} odbubskrybowano");
         }}), API_LISTENER_METHOD_NOT_ACCESSIBLE(new HashMap<Language, String>() {{
             put(Language.EN, "DiscordSRV API Listener {listenername} method {methodname} was inaccessible despite efforts to make it accessible");
+            put(Language.FI, "DiscordSRV API Listener {listenername} metodia {methodname} ei voitu saavuttaa yrityksistä huolimatta");
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:434]:
```suggestion
            put(Language.FI, "DiscordSRV API Listener {listenername} metodia {methodname} ei voitu kutsua koske ne ovat yksityisiä");
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 17]
```diff
@@ -435,6 +470,7 @@ public enum InternalMessage {
             put(Language.PL, "Nie udało się pobrać adresu URL");
         }}), PLUGIN_HOOK_ENABLING(new HashMap<Language, String>() {{
             put(Language.EN, "Enabling {plugin} hook");
+            put(Language.FI, "Käynnistetään {plugin} hook");
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:473]:
```suggestion
            put(Language.FI, "Käynnistetään {plugin} yhteensopivuus");
```

File: src/main/resources/messages/fi.yml [hunk 18]
```diff
@@ -0,0 +1,444 @@
...
+DiscordLinkedAccountRequired: "Yritit lähettää viestin pelin chat kanavalle, mutta tämä palvelin vaatii, että sinun käyttäjäsi on linkitetty Discord-tiliisi. Linkitä se pelissä kirjoittamalla `/discord link `. \n```%message%```"
+DiscordLinkedAccountCheckFailed: "Tilisi linkitys tilaa ei voitu tarkistaa. Yritä myöhemmin uudelleen."
+# Minecraft
+ClickToCopyCode: "Klikkaa kopioidaksesi"
+MinecraftAccountLinked: "&bSinun UUID:si on linkitetty Discord käyttäjään: %username% (%id%)"
+MinecraftAccountAlreadyLinked: "&bSinun Minecraft käyttäjäsi on jo liitetty Discord tiliin. Jos sinulla on tähän oikeus, voit poistaa linkityksen komennolla /discord unlink."
+LinkedCommandSuccess: "&bMinecraft käyttäjäsi on liitetty tiliin: %name%."
+UnlinkCommandSuccess: "&bMinecraft käyttäjäsi yhdistäminen on poistettu tililtä: %name%."
+MinecraftNoLinkedAccount: "&cMinecraft käyttäjäsi ei ole yhdistetty Discord tiliin."
+LinkingError: "&cTilejä ei voitu linkittää sisäisen ongelman myötä. Ota yhteyttä ylläpitoon!"
```
Comment (Vankka) [inline src/main/resources/messages/fi.yml:443]:
```suggestion
LinkingError: "&cTilejä ei voitu linkittää sisäisen virheen myötä. Ota yhteyttä ylläpitoon!"
```

File: src/main/resources/config/fi.yml [hunk 19]
```diff
@@ -0,0 +1,350 @@
...
+#
+# JDA                   - JDA:n debug viestit
+# JDA_REST_ACTIONS      - JDA:n rest action debug
+# CALLSTACKS            - Näyttää stack tracen DiscordSRV:n debugeista
+#
+# Esim. "Debug: [GROUP_SYNC, PRESENCE]"
+#
+Debug: []
+
+# vanha vaihtoehto uuteen Debug-Asetukseen siirtymistä varten
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:66]:
```suggestion
# Vanha vaihtoehto uuteen Debug-Asetukseen siirtymistä varten
```

File: src/main/resources/config/fi.yml [hunk 20]
```diff
@@ -0,0 +1,350 @@
...
+# WATCHDOG              - Watchdog ongelmat
+# BAN_SYNCHRONIZATION   - Porttikieltojen synkronointi
+# LP_CONTEXTS           - Luckperms pluginin kontekstit
+# ACCOUNT_LINKING       - discord/minecraft käyttäjä linkitys
+#
+# UNCATEGORIZED         - Mikä tahansa joka ei sovi aiemmin mainittuihin
+# ALL                   - Kaikki aiemmat kategoriat (mukaan lukien "UNCATEGORIZED")
+#
+# JDA                   - JDA:n debug viestit
+# JDA_REST_ACTIONS      - JDA:n rest action debug
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:59]:
```suggestion
# JDA_REST_ACTIONS      - JDA:n rest action debuggaus
```

File: src/main/resources/config/fi.yml [hunk 21]
```diff
@@ -0,0 +1,350 @@
...
+Experiment_WebhookChatMessageAvatarFromDiscord: false
+Experiment_WebhookChatMessageUsernameFilters: {}
+
+# Embed ja Webhook kuva/avatar url formatointi
+# Vaihtoehtoinen embed avatar url offline palvelimia varten:
+# https://cravatar.eu/helmavatar/{username}/{size}.png#{texture}
+AvatarUrl: https://crafatar.com/avatars/{uuid-nodashes}.png?size={size}&overlay#{texture}
+
+# Reserializer
+# Muuntaa formatoinnin (bold, italics, underline, strikethrough) Minecraftin ja Discordin välillä
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:93]:
```suggestion
# Muuntaa formatoinnin (lihavointi, kursivointi, alleviivaus, yliviivaus) Minecraftin ja Discordin välillä
```

File: src/main/resources/config/fi.yml [hunk 22]
```diff
@@ -0,0 +1,350 @@
...
+ForcedLanguage: none
+PrintGuildsAndChannels: true
+ForceTLSv12: true
+NoopHostnameVerifier: false
+MaximumAttemptsForSystemDNSBeforeUsingFallbackDNS: 3
+TimestampFormat: EEE, d. MMM yyyy HH:mm:ss z
+DateFormat: yyyy-MM-dd
+# https://docs.discordsrv.com/config/#Timezone
+Timezone: default
+# MinecraftMentionSound: Lähetetäänkö pelaajalle ilmoitusääni Minecraftissä kun hänet mainitaan discordissa?
```
Comment (Vankka) [inline src/main/resources/config/fi.yml:109]:
```suggestion
# MinecraftMentionSound: Lähetetäänkö pelaajalle ilmoitusääni Minecraftissä kun hänet mainitaan Discordissa?
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 23]
```diff
@@ -48,6 +49,21 @@ public class LangUtil {
     public enum InternalMessage {
 
         ASM_WARNING(new HashMap<Language, String>() {{
+			put(Language.FI,
+                    "\n" +
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:53]:
```suggestion
            put(Language.FI,
                    "\n" +
```

File: src/main/java/github/scarsz/discordsrv/util/LangUtil.java [hunk 24]
```diff
@@ -48,6 +49,21 @@ public class LangUtil {
...
+                    "\n" +
+                    "Yrität käyttää DiscordSRV:tä ASM 4 versiolla. DiscordSRV vaatii vähintään ASM 5:n toimiakseen.\n" +
+                    "DiscordSRV EI TULE TOIMIMAAN ilman ASM 5 versiota. Syytä tästä palvelin softasi kehittäjiä heidän vanhan version käyttämisestään.\n" +
+                    "\n" +
+                    "Ohjeet ASM 5 päivittämiseksi:\n" +
+                    "1. Etsi palvelimen {specialsourcefolder} kansio\n" +
+                    "2. Poista SpecialSource-1.7-SNAPSHOT.jar jar tiedosto\n" +
+                    "3. Lataa SpecialSource v1.7.4: https://repo1.maven.org/maven2/net/md-5/SpecialSource/1.7.4/SpecialSource-1.7.4.jar\n" +
+		            "4. Kopioi saatu jar tiedosto palvelimen {specialsourcefolder} kansioon, jonka löysit aiemmin.\n" +
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/util/LangUtil.java:62]:
```suggestion
                    "3. Lataa SpecialSource v1.7.4: https://repo1.maven.org/maven2/net/md-5/SpecialSource/1.7.4/SpecialSource-1.7.4.jar\n" +
                    "4. Kopioi saatu jar tiedosto palvelimen {specialsourcefolder} kansioon, jonka löysit aiemmin.\n" +
```

**Review by Vankka — COMMENTED** [review]
Inconsistent comments being Finnish & English in the same file is a bit rough, lots of minor suggestions

**Comment by granny** [comment]
Closing since OP has stated that they won't be continuing the PR. Anyone else is welcome to pick up where they left off.

**Pull request closed without merging**